experimental/features.py

```diff
@@ -0,0 +1,798 @@
+import enum
+from typing import Optional, Callable, Sequence, Tuple
+from jax import random
+from jax._src.util import prod
+from jax import numpy as np
+import jax.example_libraries.stax as ostax
+from jax import eval_shape, ShapedArray
+
+from neural_tangents import stax
+from neural_tangents._src.utils import dataclasses
+from neural_tangents._src.utils.typing import Axes
+from neural_tangents._src.stax.linear import _pool_kernel, Padding, _get_dimension_numbers, AggregateImplementation
+from neural_tangents._src.stax.linear import _Pooling as Pooling
+
+from experimental.sketching import TensorSRHT, PolyTensorSketch
+from experimental.poly_fitting import kappa0_coeffs, kappa1_coeffs, kappa0, kappa1, relu_ntk_coeffs
+""" Implementation for NTK Sketching and Random Features """
+
+
+@dataclasses.dataclass
+class Features:
+  nngp_feat: np.ndarray
+  ntk_feat: np.ndarray
+  norms: np.ndarray
+
+  is_reversed: bool = dataclasses.field(pytree_node=False)
+
+  batch_axis: int = 0
+  channel_axis: int = -1
+
+  replace = ...  # type: Callable[..., 'Features']
+
+
+class ReluFeaturesMethod(enum.Enum):
+  """Method for ReLU NNGP/NTK features approximation."""
+  RANDFEAT = 'RANDFEAT'
+  POLYSKETCH = 'POLYSKETCH'
+  PSRF = 'PSRF'
+  POLY = 'POLY'
+  EXACT = 'EXACT'
+
+
+def layer(layer_fn):
```

Comment:
To have a clear picture of the `Features` API for users, it would be nice to type-annotate its layers and `init_fn`/`feature_fn`, by analogy to

https://github.com/google/neural-tangents/blob/1e4754fb122ecab8ba2d2a3ca0379be824b4525d/neural_tangents/_src/utils/typing.py#L223

and e.g. https://github.com/google/neural-tangents/blob/1e4754fb122ecab8ba2d2a3ca0379be824b4525d/neural_tangents/_src/stax/linear.py#L223

(In your case `InternalFeatureLayer = Tuple[FeatureInitFn, FeatureFn]`, and for the functions you can define the typing protocols like https://github.com/google/neural-tangents/blob/1e4754fb122ecab8ba2d2a3ca0379be824b4525d/neural_tangents/_src/utils/typing.py#L86)